src/OrchardCore/OrchardCore.Users.Core/Services/UserService.cs

```diff
@@ -370,4 +374,56 @@ public async Task<IUser> RegisterAsync(RegisterUserForm model, Action<string, st
 
         return user;
     }
+
+    /// <inheritdoc/>
+    public async Task<bool> EnableAsync(IUser user)
+    {
+        ArgumentNullException.ThrowIfNull(user);
+
+        if (user is User u)
+        {
+            u.IsEnabled = true;
+        }
+
+        var result = await _userManager.UpdateAsync(user);
+
+        if (result.Succeeded)
+        {
+            var userContext = new UserContext(user);
+            await _handlers.InvokeAsync((handler, context) => handler.EnabledAsync(context), userContext, _logger);
+        }
+
+        return result.Succeeded;
+    }
+
+    /// <inheritdoc/>
+    public async Task<bool> DisableAsync(IUser user)
+    {
+        ArgumentNullException.ThrowIfNull(user);
+
+        var enabledUsersOfAdminRole = (await _userManager.GetUsersInRoleAsync(OrchardCoreConstants.Roles.Administrator))
+            .Cast<User>()
+            .Where(user => user.IsEnabled)
```

Comment:
```suggestion
            .Where(user => user.IsEnabled)
            .Take(2)
```

Reply:
Why do you need to take 2?

Reply:
Please let us keep things as it's. This PR takes much time

Reply:
this makes the query faster, We are doing this logic just to check if we have one or more admins. so checking for 2 will make the query run faster and gives us the same result. Take the change and I'll merge it.